js/views/playingScreen.js

```diff
@@ -203,17 +192,32 @@ var PlayingScreen = React.createClass({
   },
   
   getDirectionParams: function(pageX, pageY) {
-    var dx = pageX - this.state.xVRMouseStart;
-    var dy = pageY - this.state.yVRMouseStart;
+    var dx = pageX - this.state.xVrMouseStart;
+    var dy = pageY - this.state.yVrMouseStart;
     var maxDegreesX = 90;
     var maxDegreesY = 120;
     var degreesForPixelYaw = maxDegreesX / this.props.componentWidth;
     var degreesForPixelPitch = maxDegreesY / this.props.componentHeight;
-    var yaw = (this.props.controller.state.viewingDirection.yaw || 0) + dx * degreesForPixelYaw;
-    var pitch = (this.props.controller.state.viewingDirection.pitch || 0) + dy * degreesForPixelPitch;
+    var yaw = this.getViewingDirectionParamValue("yaw") + dx * degreesForPixelYaw;
+    var pitch = this.getViewingDirectionParamValue("pitch") + dy * degreesForPixelPitch;
     return [yaw, 0, pitch];
   },
 
+  /**
+   * @param paramName {string} "yaw", "pitch"
+   * @returns {number}
+   */
+  getViewingDirectionParamValue: function(paramName) {
+    var viewingDirectionYaw = 0;
+    if (this.props.controller &&
+      this.props.controller.state &&
+      this.props.controller.state.viewingDirection &&
+      _.isNumber(this.props.controller.state.viewingDirection[paramName])) {
```

Comment:
usage of underscore, should we change this to typeof X === "number" ?

Reply:
also, should we do a null check on this.props.controller.state.viewingDirection[paramName]?
Or typeof can do null check?

Reply:
Confirmed that typeof handles null check, so we are good.

Reply:
fix